src/ar100/timer.h

```diff
@@ -0,0 +1,21 @@
+#include <stdint.h>
+#include "asm/spr.h"
+#include "sched.h"
+
+static inline unsigned long mfspr(unsigned long add){
+    unsigned long ret;
+    __asm__ __volatile__ ("l.mfspr %0,r0,%1" : "=r" (ret) : "K" (add));
+    return ret;
+}
+
+static inline void mtspr(unsigned long add, unsigned long val){
+    __asm__ __volatile__ ("l.mtspr r0,%1,%0" : : "K" (add), "r" (val));
+}
+
+uint8_t timer_interrupt_pending(void);
```

Comment:
Same comment as gpio.h - since timer.h is used by the generic code it would be preferable to move local definitions to a local header (eg, internal.h).